src/App.jsx

```diff
@@ -14,33 +17,111 @@ export const goodsFromServer = [
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_KEYS = {
+  alphabet: 'alphabet',
+  length: 'length',
+};
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+export const App = () => {
+  const [sortBy, setSortBy] = useState(''); // lenght, alphabet
+  const [reverse, setReverse] = useState(1);
+  const [resetActive, setResetActive] = useState(false);
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  const resetButtons = () => {
+    document.querySelectorAll('.button').forEach(button => {
+      button.classList.add('is-light');
+    });
+  };
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  const sortedGoods = goodsFromServer.toSorted((goodA, goodB) => {
+    if (sortBy === 'alphabet') {
+      return goodA.localeCompare(goodB);
+    }
+
+    if (sortBy === 'length') {
+      return goodA.length - goodB.length;
+    }
+
+    return 0;
+  });
+
+  const reverseGoods = reverse < 0 ? sortedGoods.toReversed() : sortedGoods;
+  const buttonChangeClass = key => {
+    for (const button of document.querySelectorAll('.button')) {
+      if (button.dataset.sort === key) {
+        button.classList.remove('is-light');
+      } else if (button.dataset.sort) {
+        button.classList.add('is-light');
+      }
+    }
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          data-sort={SORT_KEYS.alphabet}
+          className="button is-info is-light"
+          onClick={() => {
+            buttonChangeClass(SORT_KEYS.alphabet);
+            setResetActive(true);
+            setSortBy(SORT_KEYS.alphabet);
+          }}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          data-sort={SORT_KEYS.length}
+          className="button is-success is-light"
+          onClick={() => {
+            buttonChangeClass(SORT_KEYS.length);
+            setResetActive(true);
+            setSortBy(SORT_KEYS.length);
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className="button is-warning is-light"
+          onClick={e => {
+            e.target.classList.toggle('is-light');
+            setReverse(reverse * -1);
+            if (reverse === 1) {
+              setResetActive(true);
+            } else {
+              setResetActive(false);
```

Comment:
The logic for toggling the reverse state might not work as intended. The `reverse` state is updated asynchronously, so using the current state value (`reverse`) immediately after calling `setReverse` might not reflect the updated state. Consider using a functional update to ensure the correct state is used.